lib/node/nodes/dot-density.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var dot = require('dot');
+dot.templateSettings.strip = false;
+
+var Node = require('../node');
+
+var TYPE = 'dot_density';
+var PARAMS = {
+    source: Node.PARAM.NODE(Node.GEOMETRY.POLYGON),
+    points_col_name: Node.PARAM.NULLABLE(Node.PARAM.STRING),
```

Comment:
Yeah I wanted to discuss this with you to try and figure out what the best practice is. So Dot Density can run in a number of ways. It always needs to have a geometry column but then you can either 

- Pass just a plane number of points to populate each geometry by : the points param. Example [here](https://github.com/CartoDB/camshaft/blob/b639b5e8c22cd2d999f3e38bea86cf6d80de1647/examples/viewer/examples.js#L529) 
- Pass the name of a points column, in which case Dot Density will populate each geometry with the corresponding number of points in that column. Example [here](https://github.com/CartoDB/camshaft/blob/b639b5e8c22cd2d999f3e38bea86cf6d80de1647/examples/viewer/examples.js#L550) 
- Pass the name of a points column + the name of a category column so that all points that get generated for that geometry get an additional column called category which corresponds to the category column name. Same example as before 

So in a sense we have  ```dot_density(geometry, points)```, ```dot_density(geometry, points_col_name)``` or ```dot_density(geometry,points_col_name,category_col_name)```  So we need at least one of points or points_col_name. 

The alternative would be having something like a Node.PARAM.VALUE_OR_COLUMN_NAME parameter that could accept either a value or a column name.

What do you think?